src/models/ProjectNotification.ts

```diff
@@ -26,6 +27,96 @@ export class ProjectNotification {
         await this.notifyDanglingDataValues(recipients);
     }
 
+    async notifyOnDataReady(period: string, id: string) {
+        const res = await this.api.metadata
+            .get({
+                userRoles: {
+                    fields: {
+                        id: true,
+                        users: {
+                            email: true,
+                            id: true,
+                            userGroups: {
+                                id: true,
+                                name: true,
+                            },
+                        },
+                    },
+                    filter: { name: { in: baseConfig.userRoles.dataReviewer } },
+                },
+                dataSets: {
+                    fields: {
+                        id: true,
+                        userGroupAccesses: true,
+                        userAccesses: { id: true },
+                    },
+                    filter: { id: { in: [id] } },
+                },
+            })
+            .getData();
+
+        const { displayName: user, username } = this.currentUser.data;
+
+        const subject = i18n.t("{{username}} is requesting for data review", { username });
+        const pageLink = window.location.href;
+        const dataSetType = pageLink.includes("actual") ? "ACTUAL" : "TARGET";
+        const userAccessEmails = res.userRoles[0].users
```

Comment:
TS is not complaining because we are using an older version, but array[0] is in fact `T | undefined`, not T. In other words, that would raise a runtime exception if no user roles are returned.

Anyway in this case, we don't need to get the first userRole, `userRoles.dataReviewer` is an array, so let's consider all the user roles in the response. (flatMap may come handy)